datasetinsights/io/compression.py

```diff
@@ -0,0 +1,35 @@
+import logging
+import shutil
+
+import filetype
+
+logger = logging.getLogger(__name__)
+
+
+class Compression:
+    @staticmethod
+    def decompress(filepath, destination):
+        try:
+            extension = _get_file_extension_from_filepath(filepath)
+            shutil.unpack_archive(filepath, destination, extension)
```

Comment:
I actually think this implementation is enough. I would prefer not have to maintain code such as functions defined in `_get_file_extension_from_filepath` if this is already handled by standard library, unless this method tries to do something that is very different, or some side effects are required but not implemented in standard library...

Reply:
`_get_file_extension_from_filepath ` detect file extension by reading the file header/data instead of file name which is done in standard library. Keeping the current implementation will also help us when downloading files from HTTP when we dont know the file name and file type so we can just download it without giving any extension to it and `_get_file_extension_from_filepath` will automatically detect filetype and do the necessary operation.

Reply:
Why do we have to change file extension after the file was downloaded? Shouldn't we simply take the file as is? It's very unlikely we will encounter a dataset that have incorrect filename extension. If this is the issue in GroceriesReal and SynthDet sample dataset that we have control, we should make the filename extension correct when the file is downloaded.